### ISSUE TITLE
File size of project file

### ISSUE THREAD

**Issue by tsufz** [mzmine/mzmine2#275]
Hi Tomas,
I am wondering if it would possible to implement another strategy for the storage of the project file. By now, all raw data is stored within the project. In our case this may produce maybe some ten GB project files. Hence, saving the project file gets a time consuming process hampering the whole workflow. In my opinion, linking to external raw files would be a great advantage. The result would be slim project files, the raw data files hang around anyway.

Thanks
Tobias

**Comment by tomas-pluskal** [comment]
Hi Tobias,

I agree! :)

Tomas

**Comment by tsufz** [comment]
Hi Tomas,
Great! Maybe it should be on the decision of the user. So, there should be a switch in the project settings.

Best wishes
Tobias

**Comment by tomas-pluskal** [comment]
Again, I agree. That's about as much as I can do at the moment.

**Comment by dorvan** [comment]
@tomas-pluskal we obtain a 48Gb of project file, totally unmanageable.

**Comment by tsufz** [comment]
Hi Thomas @tomas-pluskal ,
I could commit some programming ressources on this topic as I hired a good Java programmer for some months. However, I would like to know if this will be fixed in MZmine 3.0 and what is the rollout perspective?

**Comment by tomas-pluskal** [comment]
Hi @tsufz 
That sounds fantastic. I myself am planning to spend the whole August to mid-September working on MZmine 3, so I hope by that time MZmine 3 would become mature enough to start planning the release.

As for the project saving/loading update, there are two ways to go:
1) Easy approach: simply add an option to skip saving the raw data, but everything else remains the same
2) Better approach: change the project file format to contain data in standard formats (mzML for raw data and mzTab for peak lists, with the mzML data being optional). This would mean implementing better support for mzTab and new modules for project loading/saving. It would take more work, but it would be much more future-proof (thinking of MZmine 3).

**Comment by tsufz** [comment]
Hi @tomas-pluskal ,
I will talk with my programmer about the efforts. However, it would be nice if you could give some hints on which moduls we have to work with. I would like to avoid delays by searching for the correct moduls.

**Comment by tomas-pluskal** [comment]
These are the current modules for project loading and saving:

https://github.com/mzmine/mzmine2/tree/master/src/main/java/net/sf/mzmine/modules/projectmethods/projectload

https://github.com/mzmine/mzmine2/tree/master/src/main/java/net/sf/mzmine/modules/projectmethods/projectsave

In case you'd like to go for the mzML/mzTab approach, I can supervise your programmer to take the right steps.

**Comment by tsufz** [comment]
@tomas-pluskal, is there any solution for this? We get really crazy on that issue. On my new super computer, I need around 30 min for to run 40 samples with import, chromatogram building, alignment, gap filling annotation. The saving of the project requires another 1 h!

**Comment by tsufz** [comment]
PS, no chance to contribute here...

**Comment by tsufz** [comment]
We get here 40 GB files for batches of GC-Orbitrap data. Who can handle that?

**Comment by robinschmid** [comment]
Do you convert the data?
A good way to limit file size is:
- convert to mzML with these msconvert filters:
  - peak picking
  - thresholding to cut out noisy signals (absolute intensity + set noise level)

![image](https://user-images.githubusercontent.com/10366914/69362211-152cbf00-0c5c-11ea-8446-2f53e301d706.png)

**Comment by Annexhc** [comment]
like Robin suggested, make sure you are using centroid data. I have analysed GC-TOF data with MZmine. Without prior centroiding the project size gets out of hand.

**Comment by tsufz** [comment]
We use centroid data. Our batches are quite big. I don't see added value 
in storing everything in one container. Actually, this is waste of space 
and time (saving time) for me.

**Comment by tsufz** [comment]
Hi,
Might work for someone, but for us it makes no sense to truncate the 
data sets. Okay, noise at the low level, but cutting down to most 
intensive, no.

An option to save the project without raw files is really highly 
appreciated.

Yours,
Tobias

**Comment by tsufz** [comment]
Just do to explain why truncation is not appropriate for us.

We are working with environmental samples and we are doing non-target 
analysis, this is quite different from metabolomics. Thus, we want to 
see as many peaks as possible in the beginning. We cannot look on fold 
changes between control and treated sample. Actually, we have no control 
sample...

Filtering down is downstream analysis...

Hence, change your mind with this containering of raw files, please.

**Comment by robinschmid** [comment]
My approach will only cut out the noise **not** limit your scan to the maximum X signals.
What noise level do you use in MZmine's **mass detection**?

You can use this noise level to filter the scans during conversion. This reduces the size drastically, without changing your results in any bit.

For this raw file I would use ~150 as the noise level.
![image](https://user-images.githubusercontent.com/10366914/69369866-ccc8cd80-0c6a-11ea-97b0-e3c13f803253.png)

**Comment by tsufz** [comment]
Hi,
Does not work with Orbitrap data... The filter does not reduce size, 
because there is less noise by default. The system noise is reduced by 
the FT IMHO.

**Comment by robinschmid** [comment]
I have used the same approach to filter data from different orbitrap instruments. The scans are usually less noisy, however, you can still cut out noise.
Can you upload one representative raw data file and provide the noise level that you usually apply during mass detection?

Q-Exactive data:
![image](https://user-images.githubusercontent.com/10366914/69382571-64d3b080-0c85-11ea-9843-1a6dd1102bfe.png)

**Comment by tomas-pluskal** [comment]
@tsufz I agree with @robinschmid, let's look at actual data so this discussion can move forward. As for redesigning the project saving and loading in MZmine, I agree it would be useful but so far there have been zero volunteers to do it :)

**Comment by tsufz** [comment]
That's a pity. The file size and the time consuming to save is really annoying. Will it be solved in MZmine 3?

**Comment by tsufz** [comment]
Again, I cannot cut noise in our data. The filter decreases file size very minimal.

**Comment by tsufz** [comment]
My problem is that we have to open projects of PhDs and master students from time to time, we cannot run more countless projects again from the beginning, starting with exporting raw files again.

However, the noise cut off is 5E03.

**Comment by tsufz** [comment]
In meanwhile, we fix all our batch files, because all was changed to 'feature'....

**Comment by tomas-pluskal** [comment]
@tsufz What you write is not constructive at all. Again, please start by uploading one of your files so we can take a look at it. Please also understand that this is an open source project that people work on in their free time, and you are most welcome to contribute the missing features that you need.

**Comment by tsufz** [comment]
Thanks for the offer, bit we don't want to crop the files. Close #275

**Comment by tomas-pluskal** [comment]
Actually, this issue should stay open, because the problem has not been addressed.
